p2p/subscriber_metrics.go

```diff
@@ -0,0 +1,130 @@
+package p2p
+
+import (
+	"context"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+const (
+	statusKey    = "status"
+	statusAccept = "accept"
+	statusIgnore = "ignore"
+	statusReject = "reject"
+)
+
+type subscriberMetrics struct {
+	messageNumInst  metric.Int64Counter
+	messageSizeInst metric.Int64Histogram
+
+	messageTimeLastMu sync.Mutex
+	messageTimeLast   time.Time
+	messageTimeInst   metric.Float64Histogram
+
+	subscriptionNum     atomic.Int64
+	subscriptionNumInst metric.Int64ObservableGauge
+	subscriptionNumReg  metric.Registration
+}
+
+func newSubscriberMetrics() (m *subscriberMetrics, err error) {
+	m = new(subscriberMetrics)
+	m.messageNumInst, err = meter.Int64Counter(
+		"hdr_p2p_sub_msg_num",
+		metric.WithDescription("header message count"),
+	)
+	if err != nil {
+		return nil, err
+	}
+	m.messageSizeInst, err = meter.Int64Histogram(
+		"hdr_p2p_sub_msg_size",
```

Comment:
it is convinient to end name with type to make it easier to navigate in metrics explorer. So for histogram it will be "hdr_p2p_sub_msg_size_hist", for counter it will be "_counter" suffix and so on

Reply:
I saw that Prometheus automatically adds _sum, _bucket, _count

Reply:
well it could be another dev who adds those, because he is using different naming schema. And since explorer shows you all metrics in prometheus at the same time, it is just nice to have this hint for yourself. Anyway, would be great if we can me consistent with naming.

Reply:
Sure, updated them